### PR TITLE
PUB-1066 - Approval Submission screens

### PR DESCRIPTION
**Pull request by ChrisS1512**
### JIRA link ###

https://tools.hmcts.net/jira/browse/PUB-1066

### Change description ###

This PR provides the ability for users to approve a request, including the confirmation screen and radio button screen.

image

image

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/unit/views/media-account-confirmation.test.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
+import {app} from '../../../main/app';
+import request from 'supertest';
+import {expect} from 'chai';
+import sinon from 'sinon';
+import {MediaAccountApplicationService} from '../../../main/service/mediaAccountApplicationService';
+
+let htmlRes: Document;
+
+describe('Media Account Submission Page', () => {
```
Comment (furnivall) [inline src/test/unit/views/media-account-confirmation.test.ts:9]: I think this might be an error!

Comment (ChrisS1512) [reply src/test/unit/views/media-account-confirmation.test.ts]: Fixed :)

File: src/test/unit/views/media-account-confirmation.test.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    'id': '1234',
+    'fullName': 'Test Name',
+    'email': 'a@b.com',
+    'employer': 'employer',
+    'image': '12345',
+    'imageName': 'ImageName.jpg',
+    'requestDate': '09 May 2022',
+    'status': 'PENDING',
+    'statusDate': '2022-05-09T00:00:01',
+  };
```
Comment (joshblackmoor) [inline src/test/unit/views/media-account-confirmation.test.ts:47]: This dummyApplication object is repeated in a few of the test files, maybe we could add a helper file for some of the consts used in multiple files to DRY out the code a bit?

**Review by joshblackmoor — CHANGES_REQUESTED** [review]

File: src/main/controllers/MediaAccountApprovalController.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    if (applicantData) {
+      return MediaAccountApprovalController.applicationFoundFlow(req, res, approved, applicantId, applicantData);
+    }
+    res.render('error', req.i18n.getDataByLanguage(req.lng).error);
+  }
+
+  /**
+   * This handles the pages that render when submitting an approval, if the applicant has been found.
+   */
+  private static async applicationFoundFlow(req, res, approved, applicantId, applicantData): Promise<void> {
```
Comment (kurt-uk) [inline src/main/controllers/MediaAccountApprovalController.ts:39]: does this need to be async?

File: src/main/service/mediaAccountApplicationService.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    const mediaAccount = await this.getApplicationById(applicationId);
+
+    if (mediaAccount && mediaAccount.status === status) {
+      return mediaAccount;
+    }
+
+    return null;
+  }
+
+  public async getApplicationImageById(imageId): Promise<Blob> {
```
Comment (kurt-uk) [inline src/main/service/mediaAccountApplicationService.ts:33]: is this getting an application or is this getting an image?

**Review by kurt-uk — CHANGES_REQUESTED** [review]
think we might need some e2e's for this

File: src/main/controllers/MediaAccountApprovalController.ts (hunk 3 above)
Comment (ChrisS1512) [reply src/main/controllers/MediaAccountApprovalController.ts]: Removed :)

File: src/main/service/mediaAccountApplicationService.ts (hunk 4 above)
Comment (ChrisS1512) [reply src/main/service/mediaAccountApplicationService.ts]: It's getting an image.  I've updated this now to getImageById.

File: src/test/unit/views/media-account-confirmation.test.ts (hunk 2 above)
Comment (ChrisS1512) [reply src/test/unit/views/media-account-confirmation.test.ts:47]: Good point :) I've moved this into a consts file

**Review by joshblackmoor — DISMISSED** [review]
Looks good!

**Review by kurt-uk — DISMISSED** [review]

**Review by furnivall — APPROVED** [review]

File: src/main/views/media-account-approval-confirmation.njk [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+{% block beforeContent %}
+  {{ super() }}
+{% endblock %}
+{% block content %}
+  <div class="govuk-panel govuk-panel--confirmation">
+    <div class="govuk-panel__body">
+        {{ panelTitle }}
+    </div>
+  </div>
+  {{ govukTable({
```
[the remark this answers is not in the document]
Comment (ChrisS1512) [reply src/main/views/media-account-approval-confirmation.njk:18]: I'll have a look at this as part of 1065 and change them on all pages if possible :)

**Pull request merged**